### PR TITLE
[docs] Update documentation structure with markdown files

### PR DESCRIPTION
**Pull request by mugulmd**
## Description

This PR aims at updating the project documentation on several levels: 
- with a new doxygen setup to build a HTML documentation of the C++ code
- with a new `README.md` file in several sub-directories of the `aliceVision` directory so that users can quickly understand how the module is structured and how to use it
- with a `BIBLIOGRAPHY.md` file to store the project bibliography.

### CODE REVIEW

File: docs/doxygen/Doxyfile.in [hunk 1]
```diff
@@ -1509,19 +1755,14 @@ EXT_LINKS_IN_WINDOW    = NO
...
+# The FORMULA_MACROFILE can contain LaTeX \newcommand and \renewcommand commands
+# to create new LaTeX commands to be used in formulas as building blocks. See
+# the section "Including formulas" for details.
 
-FORMULA_TRANSPARENT    = YES
+FORMULA_MACROFILE      =
 
 # Enable the USE_MATHJAX option to render LaTeX formulas using MathJax (see
-# http://www.mathjax.org) which uses client side Javascript for the rendering
+# https://www.mathjax.org) which uses client side JavaScript for the rendering
```
Comment (simogasp) [inline docs/doxygen/Doxyfile.in:1765]: maybe enable `USE_MATHJAX`, the rendering should be nicer

File: src/aliceVision/robustEstimation/README.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+- Too large: Models are contaminated by outliers (false data).
+
+AC-Ransac [ACRANSAC] uses the a contrario methodology in order to find a model that best fits the
+data with a confidence threshold T that adapts automatically to noise. It so finds a model and its
+associated noise. If there is too much noise, the a contrario method returns that no model was
+found.
+
+![A contrario robust estimation, noise adaptivity.](../../../docs/img/ACRansac.png)
+
+Here an example of how to find a best fit line, by using the a contrario robust estimation framework:
```
Comment (simogasp) [inline src/aliceVision/robustEstimation/README.md:98]:
```suggestion
Here an example of how to find a best fitting line, by using the a contrario robust estimation framework:
```

File: src/aliceVision/sfm/README.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
+# sfm
+
+sfm is the module related to Structure from Motion.
+It handles storage of SfM related data and method to solve SfM problems (camera pose estimation, structure triangulation, bundle adjustment).
+
+
+## A generic SfM data container
```
Comment (simogasp) [inline src/aliceVision/sfm/README.md:7]: this should go in the sfmData module, shouldn't it?

Comment (mugulmd) [reply src/aliceVision/sfm/README.md:7]: You're right!

**Review by fabiencastan — APPROVED** [review]

**Pull request merged**